src/api/operator/operatorApi.ts

```diff
@@ -178,13 +178,19 @@ export async function getOrder(params: GetOrderParams): Promise<RawOrder | null>
     if (response.status === 404) {
       return null
     }
-    throw new Error(`Request failed: [${response.status}] ${await response.text()}`)
+    // Just in case response.text() fails
+    const responseText = await response.text().catch((e) => {
```

Comment:
this will still be uncaught as it's not in the scope of a `try/catch` block, this is is why i recommended moving it up into the prvious `try/catch` block

```js
 let prom2 = await fakePromise(3000, true).catch(e => { throw new Error(e) })
// VM513:2 Uncaught Error: ERROR
//    at <anonymous>:1:63
//    at async <anonymous>:1
```

Reply:
just set the error message u want the `catch` block to use inside the inlined `throw new Error(<message>)` or just `throw "<message>"` and you can just have the `catch block` rethrow/log the error passed from the `try` throw

Reply:
Well, I don't think is uncought. The function is asyn, so it just fails with the right message

Reply:
I want this to not be caught by the first try/catch. Besides, that statement also throws.
The only cases where it doesn't throw are when:
1. There is a valid response
2. The order doesn't exist (404) which then it returns `null`

Otherwise, it's a failure, and I want the caller to deal with it.